src/actions/resize.ts

```diff
@@ -70,8 +70,8 @@ import {ResizeFillAction} from "./resize/ResizeFillAction";
  * @param {number|string} height The required height of a transformed asset.
  * @return {Actions.Resize.ScaleAction}
  */
-function scale(width?: number | string, height?: number | string): ScaleAction {
-  return new ScaleAction('scale', width, height);
+function scale(width?: number | string, height?: number | string): { __EXTENDED__:true } & ScaleAction {
```

Comment:
what is __EXTENDED__ ?

Reply:
Just a private key (can be named anything), to ensure that this return type is NOT just new ScaleAction.

This ensures that `resize(new ScaleAction())` is a type error, while `resize(scale())` is not a type error

Reply:
ok i see, how does the hint look like in the ide after this change?

Reply:
Good in some areas, bad in others.

In typescript, the IDE hints are amazing, but the type errors are shown in a very bad way. 
This doesn't look like a very good solution unfortunately.